playground/src/hoc/with-error-boundary.usage.tsx

```diff
@@ -1,16 +1,33 @@
 import * as React from 'react';
+import {useState} from 'react';
 
 import { withErrorBoundary } from '../hoc';
 import { ErrorMessage } from '../components';
 
 const ErrorMessageWithErrorBoundary =
   withErrorBoundary(ErrorMessage);
 
-const BrokenButton = () => (
-  <button type="button" onClick={() => { throw new Error(`Catch me!`); }}>
-    {`Throw nasty error`}
-  </button >
-);
+const BrokenButton = () => {
+  const [throwError, setThrowError] = useState(false);
+
+  if (throwError) {
+    throw new Error('Catch me!');
+  }
+
+  return (
+    <button
+      type="button"
+      onClick={() => {
+        // We don't just throw the error here, because error boundaries
```

Comment:
I would prefer just: 
> Error boundaries catch errors during rendering and do not work inside event handlers: <link>

Reply:
Tbh, I think that comment will just confuse whoever is reading through the example. I removed the comment altogether now.

Reply:
Yeah that's fine 👍